CHANGELOG.md

```diff
@@ -1,5 +1,51 @@
 # go-ipfs changelog
 
+### 0.3.10 - 2015-12-07
+
+This patch update introduces the 'ipfs update' command which will be used for
+future ipfs updates along with a few other bugfixes and documentation
+improvements.
+
+* Features
+  * support for 'ipfs update' to call external binary (@whyrusleeping)
+  * cache ipns entries to speed things up a little (@whyrusleeping)
+  * add option to version command to print repo version (@whyrusleeping)
+  * Add in some more notifications to help profile queries (@whyrusleeping)
+  * gateway: add path prefix for directory listings (@lgierth)
+  * gateway: add CurrentCommit to /version (@lgierth)
+
+* BugFixes
+  * set data and links nil if not present (@whyrusleeping)
+  * fix log hanging issue, and implement close-notify for commands (@whyrusleeping)
+  * fix dial backoff (@whyrusleeping)
+  * proper ndjson implementation (@whyrusleeping)
+  * seccat: fix secio context (@lgierth)
+  * Add newline to end of the output for a few commands. (Nick Haman)
+  * Add fixed period repo GC + test (@rht)
```

Comment:
this is a big change, may want to add a warning at the top:
> Warning: this commit introduces fixed-period repo gc, which will trigger gc after a fixed period of time. This feature is disabled by default, but can be enabled with `ipfs daemon --enable-gc`

@rht wonder if we should make this `--periodic-gc`?